src/Chapter2.hs

```diff
@@ -727,7 +735,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat (map (\x -> replicate x x) l)
```

Comment:
If you to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests :slightly_smiling_face:

Reply:
Solved with eta-recution! :smile: